R/model_eligibility.R

```diff
@@ -504,3 +527,219 @@ calc_baseline_check <- function(
+
+  if (!is.null(sd_check_table_path) | !is.null(sd_check_plot_path) | show_stats) {
+    el_detail <- eligibility %>% 
+    dplyr::left_join(lookback_stats, by = "location") %>% 
+    dplyr::left_join(fips_codes)
+  }
+
+  if (!is.null(sd_check_table_path)) {
+    tab <- el_detail %>%
+    dplyr::filter(sd_eligibility != "eligible") %>% 
+    transmute(location, location_name, model,
+      `mean ahead` = round(mean_ahead),
+      `past mean` = round(past_mean),
+      SD = round(sd, 1),
+      cutoff = round(past_mean - num_sd*sd,1),
+      `reason excluded` = sd_eligibility)
+    write_csv(tab, file = paste0(sd_check_table_path,
+     "sd_eligibility_",
+     el_detail$forecast_week_end_date[1],
+     ".csv"))
+  }
+
+  if (!is.null(sd_check_plot_path)) { 
+
+    el_detail$location_name <- factor(el_detail$location_name,
+      levels = c("US", sort(unique(el_detail$location_name[el_detail$location_name!="US"]))))
+
+    p_point <- ggplot(el_detail, aes(x = model, y = mean_ahead, color = model)) +
+    geom_point() +
+    geom_hline(aes(yintercept = past_mean), alpha = .4) +
+    geom_hline(aes(yintercept = past_mean - num_sd*sd), linetype = "dashed") +
+    geom_hline(aes(yintercept = past_mean + num_sd*sd), linetype = "dashed") +
+    facet_wrap(~location_name, scales = "free_y") +
+    theme_bw() +
+    theme(axis.title.x = element_blank(),
+      axis.title.y = element_blank(),
+      axis.text.x = element_blank(),
+      axis.ticks.x = element_blank(),
+      legend.position = "bottom")
+    ggsave(paste0(sd_check_plot_path, 
+      "exclusions_point_", 
+      el_detail$forecast_week_end_date[1], 
+      ".pdf"), 
+    plot = p_point,
+    width = 24, 
+    height = 14)
+
+    dat <- lookback_data %>% 
+    dplyr::mutate(
+      model = "observed", 
+      target_end_date = as.Date(target_end_date), 
+      value = observed) %>% 
+    dplyr::bind_rows(as.data.frame(qfm)) %>% 
+    dplyr::left_join(fips_codes)
+    dat$location_name <- factor(dat$location_name,
+      levels = c("US", sort(unique(dat$location_name[dat$location_name!="US"]))))
+    p_line <- ggplot(dat, aes(x = target_end_date, y = value, color = model)) +
+    geom_line() +
+    geom_hline(data = el_detail, aes(yintercept = past_mean), alpha = .4) +
+    geom_hline(data = el_detail, aes(yintercept = past_mean - num_sd*sd), linetype = "dashed") +
+    geom_hline(data = el_detail, aes(yintercept = past_mean + num_sd*sd), linetype = "dashed") +
+    facet_wrap(~location_name, scales = "free_y") +
+    theme_bw() +
+    theme(axis.title.x = element_blank(),
+      axis.title.y = element_blank(),
+      axis.text.x = element_blank(),
+      axis.ticks.x = element_blank(),
+      legend.position = "bottom")
+    ggsave(paste0(sd_check_plot_path, 
+      "exclusions_line_", 
+      el_detail$forecast_week_end_date[1], 
+      ".pdf"), 
+    plot = p_line,
+    width = 24, 
+    height = 14)  
+  }
+
+  if (show_stats) return(el_detail)
+
+  return(eligibility)
```

Comment:
let's make it so that the `mean_ahead` column is not saved in results -- I think an appropriate way to achieve that would be by removing it from `eligibility` before returning
```suggestion
  return(eligibility %>% dplyr::select(-mean_ahead))
```

Reply:
I'll also remove `target_end_date` at this point so I can leave it in earlier and get the date endings for the file names I meant them to have.